src/Dapr.Client/ConfigurationItem.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections.Generic;
+
+namespace Dapr.Client
+{
+    /// <summary>
+    /// Class that represents an item fetched from the Dapr Configuration API.
+    /// </summary>
+    public class ConfigurationItem
+    {
+        private readonly string key;
+        private readonly string value;
+        private readonly string version;
+        private readonly IReadOnlyDictionary<string, string> metadata;
+
+        /// <summary>
+        /// Constructor for a ConfigurationItem.
+        /// </summary>
+        /// <param name="key">The key of the configuration item.</param>
+        /// <param name="value">The value of the configuration item.</param>
+        /// <param name="version">The version of the fetched item.</param>
+        /// <param name="metadata">The metadata associated with the request.</param>
+        public ConfigurationItem(string key, string value, string version, IReadOnlyDictionary<string, string> metadata)
+        {
+            this.key = key;
+            this.value = value;
+            this.version = version;
+            this.metadata = metadata;
+        }
+
+        /// <summary>
+        /// The name of the configuration item.
+        /// </summary>
+        public string Key
```

Comment:
By the way you want to simplify these can be autoprops. `public string Key { get; }`